pandas/_libs/index_class_helper.pxi.in

```diff
@@ -77,4 +77,14 @@ cdef class {{name}}Engine(IndexEngine):
 
         return self._unpack_bool_indexer(indexer, val)
 
+    {{if name in {'Float64', 'Float32'} }}
+    def get_stargets(self, ndarray targets):
+        stargets = set(targets)
+        # account for nans
+        if -1 not in stargets and np.isnan(targets).any():
+            stargets.add(-1)
```

Comment:
why are you using -1 instead of just a canonical np.nan object?

Reply:
I like the idea of using a canonical nan-object more than, replacing a nan by -1. Because, what happens if we have nans and -1s? This uses the fact, that python first compares the identity of objects and only then the objects, so `np.nan == np.nan` because it is the same object.